cmake/add_enclave.cmake

```diff
@@ -164,9 +164,11 @@ macro(add_enclave_optee)
   set(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS)
   set(CMAKE_EXE_EXPORTS_C_FLAG)
 
-  string(REPLACE "gcc" "ld" LINKER ${CMAKE_C_COMPILER})
-  set(CMAKE_C_LINK_EXECUTABLE "${LINKER} <CMAKE_CXX_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> <LINK_LIBRARIES> -lgcc -o <TARGET>")
-  set(CMAKE_CXX_LINK_EXECUTABLE "${LINKER} <CMAKE_CXX_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> <LINK_LIBRARIES> -lgcc -o <TARGET>")
+  string(REPLACE "gcc" "ld" C_LINKER ${CMAKE_C_COMPILER})
+  string(REPLACE "g++" "ld" CXX_LINKER ${CMAKE_CXX_COMPILER})
+
+  set(CMAKE_C_LINK_EXECUTABLE "${C_LINKER} <CMAKE_C_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> <LINK_LIBRARIES> -lgcc -o <TARGET>")
+  set(CMAKE_CXX_LINK_EXECUTABLE "${CXX_LINKER} <CMAKE_CXX_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> <LINK_LIBRARIES> -lgcc -o <TARGET>")
```

Comment:
Just realized this change is kinda pointless. The only necessary change was changing CMAKE_CXX_LINK_FLAGS to CMAKE_C_LINK_FLAGS on line 170.

I'll revert once bors finishes